crates/apollo-compiler/tests/extensions.rs

```diff
@@ -34,6 +34,42 @@ fn test_orphan_extensions() {
     schema2.validate().unwrap();
 }
 
+#[test]
+fn test_orphan_extensions_schema_with_default_query_name() {
+    let input = r#"
+        extend schema { query: Query }
+        type Query { x: Int }
+    "#;
+
+    let schema = Schema::builder()
+        .adopt_orphan_extensions()
+        .parse(input, "schema.graphql")
+        .build()
+        .unwrap();
+
+    schema.validate().unwrap();
+}
+
+#[test]
+fn test_orphan_extensions_schema_def_with_extensions() {
+    let input = r#"
+        extend schema { query: Query }
+        extend schema { subscription: S }
+        schema { mutation: Mutation }
+        type Query { x: Int }
+        type Mutation { y: Int }
+        type S { z: Int }
```

Comment:
what do we expect in case of something like this, where we are extending with an implicit root operation, but it's already defined and we are using `adopt_orphan_extensions`? I want to make sure there is a validation error for this case.

```graphql
        extend schema { query: Query }
        extend schema { subscription: S }
        schema { mutation: Mutation, query: anotherQuery }
        type Query { x: Int }
        type Mutation { y: Int }
        type S { z: Int }
        type AnotherQuery { a: String }
```

Reply:
Good addition! f47aaf44bf7a1617f3a9d562b6af2f4aa55a5bc4